### PR TITLE
Refactor bootstrapping.

### PR DESCRIPTION
**Pull request by kwoodson**
The purpose of this PR is to separate parts of the bootstrapping into smaller chunks so that they can be run independently.

Changes:
- playbooks/aws/openshift-cluster/provision_custom_nodes.yml - Created in case customers want to define their own custom node types
- role/openshift_aws/tasks/provision_custom_nodes.yml - Script that handles the actual creation of the custom nodes
- moved the node-config configmap creation to its own separate role (openshift_node_bootstrap_configmap).  This could stay in master as its own separate file and be able to call it on its own but I feel like this should be its own role.
- Moved the actual master config bootstrap settings to their own file so they can be done independently

### CODE REVIEW

File: playbooks/common/openshift-master/config.yml [hunk 1]
```diff
@@ -212,6 +212,11 @@
       tasks_from: master
     when: openshift_use_kuryr | default(false) | bool
 
+  - include_role:
```
Comment (michaelgugino) [inline playbooks/common/openshift-master/config.yml]: We've already had one issue filed with users being confused about the bootstrap code being put in-line with normal provisioning.

We should add a comment to anything that references bootstrapping to clarify to users that this is not meant to be used during the normal install.

File: roles/openshift_node_bootstrap_configmap/templates/node-config.yaml.j2 [hunk 2]
```diff
@@ -0,0 +1,53 @@
+allowDisabledDocker: false
+apiVersion: v1
+authConfig:
+  authenticationCacheSize: 1000
+  authenticationCacheTTL: 5m
+  authorizationCacheSize: 1000
+  authorizationCacheTTL: 5m
+dnsBindAddress: "127.0.0.1:53"
+dnsDomain: cluster.local
+dnsIP: 0.0.0.0
```
Comment (kwoodson) [inline roles/openshift_node_bootstrap_configmap/templates/node-config.yaml.j2]: @smarterclayton, @michaelgugino, I moved the generation from a dynamic call to a template.  This was a change that went in https://github.com/openshift/openshift-ansible/pull/5924/files so I am moving it to the template.

Comment (kwoodson) [reply roles/openshift_node_bootstrap_configmap/templates/node-config.yaml.j2]: This still supports post processing for which is the happy median until we reach some understanding of config template or template + yedit.

File: roles/openshift_node_bootstrap_configmap/templates/node-config.yaml.j2 [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  - {{ openshift_node_bootstrap_configmap_cloud_provider }}
+  node-labels: {{ openshift_node_bootstrap_configmap_default_labels | union(openshift_node_bootstrap_configmap_custom_labels) | list | to_json }}
+masterClientConnectionOverrides:
+  acceptContentTypes: application/vnd.kubernetes.protobuf,application/json
+  burst: 40
+  contentType: application/vnd.kubernetes.protobuf
+  qps: 20
+masterKubeConfig: node.kubeconfig
+networkConfig:
+  mtu: "{{ openshift_node_bootstrap_configmap_network_mtu }}"
```
Comment (smarterclayton) [inline roles/openshift_node_bootstrap_configmap/templates/node-config.yaml.j2]: https://github.com/openshift/openshift-ansible/pull/5972 does similar except it uses `openshift_node_sdn_mtu` which is the customer facing value today.  Can update here as needed.

Comment (kwoodson) [reply roles/openshift_node_bootstrap_configmap/templates/node-config.yaml.j2]: @smarterclayton, this does as well but I transfer that value into the role version here: https://github.com/openshift/openshift-ansible/pull/5923/files#diff-e8f255754ee97259ae8e087cba95130aR15

**Comment by kwoodson** [comment]
closing in favor of https://github.com/openshift/openshift-ansible/pull/6006

**Pull request closed without merging**